modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/AttachmentResourceTest.java

```diff
@@ -332,6 +194,77 @@ protected Attachment testDeleteAttachment_addAttachment() throws Exception {
 			_cProduct.getExternalReferenceCode(), randomAttachment());
 	}
 
+	@Override
+	protected Attachment
+			testGetProductByExternalReferenceCodeAttachmentsPage_addAttachment(
+				String externalReferenceCode, Attachment attachment)
+		throws Exception {
+
+		return attachmentResource.postProductIdAttachment(
+			_cProduct.getCProductId(), randomAttachment());
+	}
+
+	@Override
+	protected String
+			testGetProductByExternalReferenceCodeAttachmentsPage_getExternalReferenceCode()
+		throws Exception {
+
+		return _cProduct.getExternalReferenceCode();
+	}
+
+	@Override
+	protected Attachment
+			testGetProductByExternalReferenceCodeImagesPage_addAttachment(
+				String externalReferenceCode, Attachment attachment)
+		throws Exception {
+
+		return attachmentResource.postProductIdImage(
```

Comment:
I think here is better to use the postProductByExternalReferenceCodeAttachment using the externalReferenceCode parameter and the attachment parameter (which is the difference between an attachment and an imageAttachment?)

Reply:
The difference between attachment and imageAttachment is the type field will be ContentTypes.IMAGE_JPEG in the case of imageAttachment. I think here we can't use the attachment from the parameter, because the generated method calls the randomAttachment() method, but it won't create an imageAttachment (it will create a txt file) so the test will fail, because there won't be 2 image attachments added to the Product if we use the attachment parameter.